### DISCUSSION TITLE
Emacs installation error rpl symbol(s) not found for architecture arm64

### DISCUSSION THREAD

**Discussion by Inc0n** [d12frosted/homebrew-emacs-plus#472] [Q&A] [answered]
I am specifically on Mac m1, I had not encountered an issue such as this before.
The rpl package installed from brew did not help (`brew install rpl`).
What am I missing for the installation?
The error message from `brew install emacs-plus`
```
==> ./autogen.sh
==> ./configure --enable-locallisppath=/opt/homebrew/share/emacs/site-lisp --infodir=/opt/homebrew/C
==> gmake
Last 15 lines from /Users/xception/Library/Logs/Homebrew/emacs-plus@28/03.gmake:
  "_rpl_optind", referenced from:
      _main in etags-2a6a84.o
  "_rpl_re_compile_pattern", referenced from:
      _add_regex in etags-2a6a84.o
  "_rpl_re_match", referenced from:
      _readline in etags-2a6a84.o
  "_rpl_re_search", referenced from:
      _find_entries in etags-2a6a84.o
  "_rpl_re_set_syntax", referenced from:
      _add_regex in etags-2a6a84.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[1]: *** [Makefile:376: etags] Error 1
gmake[1]: Leaving directory '/private/tmp/emacs-plusA28-20220529-14653-1f9wh2r/emacs-28.1/lib-src'
gmake: *** [Makefile:436: lib-src] Error 2
```

EDIT: As per commented in https://github.com/d12frosted/homebrew-emacs-plus/discussions/472#discussioncomment-2843384, now with `--verbose`, the log files:
There are several other log files, as echoed out with `--verbose`: under `~/Library/Logs/Homebrew/emacs-plus@28/`.
- [07.gmake.txt](https://github.com/d12frosted/homebrew-emacs-plus/files/8793241/07.gmake.txt), close to the bottom, it shows a lot more undefined symbols such as  `_c_strcasecmp`, `_c_isalnum`.
   I now think my case is similar to https://lists.gnu.org/archive/html/help-gnu-emacs/2019-02/msg00243.html, unfortunately, there was not a solution recorded in that archive.
- Most notably in `config.log` there are a lot of header files missing such as `<sys/vfs.h>`, `<sys/vfs.h>`. I am uncertain if they amount to any issues at all, also this `config.log` is fairly big.

**Comment by d12frosted** [comment]
🤔 I am also on M1 and have not encounter this error. I just tried to reinstall and everything compiled well. Can you please share full log with `--verbose` flag? You can upload file to your original post :)

**Reply by Inc0n** [reply]
Thank you, I have uploaded gmake log, and what I noticed in `config.log` in my original post.
Thank you again for taking your time, and please tell me if any other logs that I should be supplying instead.
🙏

**Reply by d12frosted** [reply]
Ah, can you please provide output from brew install itself? There are some moving bits before gmake that I would love to see as well.

**Reply by Inc0n** [reply]
Yes of course, my bad, here it is: [output.txt](https://github.com/d12frosted/homebrew-emacs-plus/files/8793742/output.txt).

**Reply by d12frosted** [reply]
Based on your output I assume you are building with some custom options, can you please provide them as well? For example, I suspect that you are building with imagemagick and dbus. Also, it worths trying to build without any options (except for icons as they do nothing special about deps and path).

**Reply by d12frosted** [reply]
🤔 On the other hand, I see `--without-dbus --without-imagemagick` in the `config` invocation. So at least confirm that you are building without options if that is the case :)

**Reply by Inc0n** [reply]
Yh my custom options are `--with-no-titlebar --with-native-comp --with-dragon-icon`, and just tried without these options and no luck for that (error still creeps in).

Thanks for digging it through for me, right now I have installed `emacs-mac` the mac port, so at least I can run an instance of emacs on my laptop, but I will miss the dragon icon for sure :)

**Comment by Andre0991** [comment, marked as answer]
Hey folks.
I was facing the same error on my _intel_ mac.

I found [1], which indicated that the issue could be the `ar` version.
I removed the `binutils` installed with homebrew and after that the compilation worked.

[1]: https://lists.gnu.org/archive/html/help-gnu-emacs/2019-02/msg00300.html

**Reply by Inc0n** [reply]
Oh yh, this solved it for me as well 👍

**Reply by d12frosted** [reply]
Oh, this is interesting. Thanks for sharing!
